packages/utils/src/lib/report.ts

```diff
@@ -56,3 +64,47 @@ export function compareIssueSeverity(
 export function sumRefs(refs: CategoryRef[]) {
   return refs.reduce((sum, { weight }) => sum + weight, 0);
 }
+
+export async function loadReports(options: PersistConfig) {
+  const { outputDir, filename, format } = options;
+  await ensureDirectoryExists(outputDir);
+
+  const dirResult = await readdir(outputDir);
+
+  const result = await Promise.allSettled(
+    dirResult
+      // filter by file extension
+      .filter(file =>
+        format ? format.find(ext => file.endsWith(`.${ext}`)) : true,
+      )
+      // filter by file name
+      .filter(file =>
+        filename
+          ? file.includes(filename)
+          : new RegExp(REPORT_NAME_PATTERN).test(file),
+      )
+      .map(file => getFileResult(file, outputDir)),
+  );
+  return result;
+}
+
+export async function loadReport(
+  options: Required<Pick<PersistConfig, 'outputDir' | 'filename'>> & {
+    format: Format;
+  },
+) {
+  const { outputDir, filename, format } = options;
+  await ensureDirectoryExists(outputDir);
+  const filePath = join(outputDir, `${filename}.${format}`);
+  const content = await readFile(filePath, 'utf8');
```

Comment:
We already have a `readJsonFile` function which would make perfect sense here.

Reply:
used it + the readTextFile